pvt/helpers/src/models/vault/VaultDeployer.ts

```diff
@@ -19,15 +20,40 @@ export default {
     const { from, mocked } = deployment;
     if (!admin) admin = from || (await ethers.getSigners())[0];
 
-    const authorizer = await this._deployAuthorizer(admin, from);
+    // Needed so that the getAuthorizerAdaptor() call in the Authorizer constructor will not revert
+    const mockEntrypoint = await deploy('v2-vault/MockAuthorizerAdaptorEntrypoint');
+
+    // Deploy the Vault with a placeholder authorizer
+    const authorizer = await this._deployAuthorizer(admin, mockEntrypoint, from);
```

Comment:
While this sequence is fine and it'll work, it's a bit different from what we'll do in production.

Namely, the entrypoint came afterwards, and in the starting point we had an old authorizer that didn't need the entrypoint.

Again, this is not a strong opinion, but we could also consider deploying a mock authorizer first with no entrypoint, and then change it to the actual authorizer with the actual entrypoint. With that approach, the first mock authorizer would be analogous to the current `Authorizer` used by the Vault in mainnet.

Something like this https://github.com/balancer-labs/balancer-v2-monorepo/pull/1977/commits/6d8f3e2805321ca0f362bb07534f7c80bd9ef178#diff-d1f1f09715847682508eabc6c0e80ebd2772af929faeacaabf0e1382e242cc9f.

Reply:
That's fine, and I like the comments too. If you can run the full suite and everything passes, I'd say you could just push that.